include/z64actor.h

```diff
@@ -581,4 +578,43 @@ typedef struct {
     /* 0x24 */ char unk_24[0x4];
 } NpcInteractInfo; // size = 0x28
 
+// Converts a number of bits to a bitmask, helper for params macros
+// e.g. 3 becomes 0b111 (7)
+#define NBITS_TO_MASK(n) \
+    ((1 << (n)) - 1)
+
+// Extracts the `n`-bit value at position `s` in `p`, shifts then masks
+// No possibility of sign extension
+#define PARAMS_GET(p, s, n) \
+    (((p) >> (s)) & NBITS_TO_MASK(n))
+
+// Extracts the `n`-bit value at position `s` in `p`, masks then shifts
+// Possibility of sign extension
+#define PARAMS_GET2(p, s, n) \
```

Comment:
More so just a thought, not how good or bad it really is. Wonder if it would be better to name this `PARAMS_GET_SIGN` instead to show it has the possibly to sign extend?

Reply:
I was thinking of naming the variants `GET_SIGNED` and `GET_UNSIGNED`, or `GET_S` and `GET_U` and keep the comment to keep them relatively short.

Reply:
im fine with the whole word, personally

Reply:
Yeah whole word seems fine, especially since most if not all usages of these are going to be in headers anyways behind other macros.

Reply:
I'd prefer using signed/unsigned over calling one "2"

and would be fine with either `GET_SIGNED`/`GET_UNSIGNED` or `GETS`/`GETU` or `GET_S`/`GET_U`